spec/integration/mac_vmi_spec.rb

```diff
@@ -0,0 +1,103 @@
+require_relative './shared_context'
+
+describe Fog::Kubevirt::Compute do
+  before :all do
+    conn = KubevirtConnection.new()
+    @client = conn.client
+    @watch = nil
+  end
+
+  it 'creates VMI and checks MAC' do
+    vm_name = 'test'
+    cpus = 1
+    memory_size = 64
+
+    volume = Fog::Kubevirt::Compute::Volume.new
+    volume.type = 'containerDisk'
+    volume.name = 'test-disk-01'
+    volume.info = 'registry:5000/kubevirt/cirros-container-disk-demo:devel'
+
+    @client.vms.create(vm_name: vm_name, cpus: cpus, memory_size: memory_size, volumes: [volume])
+
+    vm = @client.vms.get(vm_name)
+    assert_equal(vm.cpu_cores, cpus)
+    assert_equal(vm.memory, "#{memory_size}M")
+
+    vm.start()
+    # allow vmi to be created
+    sleep(1)
```

Comment:
maybe `sleep(1) until vmi_created` and let `vmi_created` check if it was actually created ?

Reply:
I empirically proved that when getting a `vmi` collection I get `resource_version`. Without the wait sometimes it is empty but in slower envs it would take more time. Will add `get`.